api/src/test/java/org/pmiops/workbench/api/DiskAdminControllerTest.java

```diff
@@ -0,0 +1,89 @@
+import static org.mockito.Mockito.when;
+import static org.pmiops.workbench.utils.TestMockFactory.createAppDisk;
+import static org.pmiops.workbench.utils.TestMockFactory.createRuntimeDisk;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.disks.DiskService;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.model.AppType;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.model.DiskStatus;
+import org.pmiops.workbench.model.ListDisksResponse;
+import org.springframework.http.ResponseEntity;
+
+@ExtendWith(MockitoExtension.class)
+public class DiskAdminControllerTest {
+  private static final String WORKSPACE_NS = "workspace-ns";
+  private static final String LOGGED_IN_USER_EMAIL = "bob@gmail.com";
+  private static final Instant NOW = Instant.now();
+
+  private static DbUser user = new DbUser();
+
+  @Mock DiskService mockDiskService;
+
+  @InjectMocks DiskAdminController diskAdminController;
+
+  @BeforeEach
+  public void setUp() {
+    user = new DbUser();
+    user.setUsername(LOGGED_IN_USER_EMAIL);
+    user.setUserId(123L);
+  }
+
+  @Test
+  public void listDisksInWorkspace() {
+    Disk rStudioDisk =
+        createAppDisk(
+            user.generatePDNameForUserApps(AppType.RSTUDIO),
+            DiskStatus.DELETING,
+            NOW.minusMillis(100).toString(),
+            user,
+            AppType.RSTUDIO);
+
+    Disk cromwellDisk =
+        createAppDisk(
+            user.generatePDNameForUserApps(AppType.CROMWELL),
+            DiskStatus.READY,
+            NOW.toString(),
+            user,
+            AppType.CROMWELL);
+
+    Disk jupyerDisk =
+        createRuntimeDisk(
+            user.generatePDName(),
+            DiskStatus.READY,
+            NOW.toString(),
+            user);
+
+    List<Disk> serviceResponse =
+        new ArrayList<>(Arrays.asList(rStudioDisk, cromwellDisk, jupyerDisk));
+
+    when(mockDiskService.findByWorkspaceNamespace(WORKSPACE_NS)).thenReturn(serviceResponse);
+
+    ResponseEntity<ListDisksResponse> response = diskAdminController.listDisksInWorkspace(WORKSPACE_NS);
+    assertThat(response.getBody())
+        .containsExactly(rStudioDisk, cromwellDisk, jupyerDisk);
+    assertThat(response.getStatusCodeValue()).isEqualTo(200);
+  }
+
+  @Test
+  public void listDisksInWorkspace_workspaceNotFound(){
+    when(mockDiskService.findByWorkspaceNamespace(WORKSPACE_NS)).thenThrow( new NotFoundException("Workspace not found: " + WORKSPACE_NS));
+    assertThrows(NotFoundException.class, () -> diskAdminController.listDisksInWorkspace(WORKSPACE_NS));
+  }
```

Comment:
1: I don't think this test is needed/useful. 
Mock subscream service let it throw an exception then we verify this class not catch then swallow it? 
2: I don't think it ever happens, seems we just return empty of disks instead of throw exception, and it make sense.  
In general, for single entity get,  e.g. `getDiskByName`, we can throw 404 it entity not found by given name. 
For list, we should return empty if no entity found

Reply:
I agree, but @jmthibault79, @Peter-Lavigne , and I were talking this morning and couldn't come to a consensus. I will delete it for now. I setup a call tomorrow so that the team can talk about the Controller/Service/Repository pattern. I'm hoping that during that call we can discuss how/if we want to handle exceptions by dependencies.

Reply:
SG. I don't think 404 is the correct use. My understanding is you expect to find something but not found, e.g. findById, findByName. If we expect to see nothing get returned, 200 with empty response makes more sense

Reply:
I forgot to delete before merging. 

To clarify, this isn't the case when we do not find any disks. This is the case when you ask for disks from a workspace that does not disks. This is saying the workspace was not found.

Reply:
ah. I see, thanks for clarifying. 
In that case, 404 is not a good response and confusing, that would indicate Disk not found since the get endpoint is for disk. 
400 also make sense meaning the workspace id is a bad request? Or 500

Reply:
Hmmm, that is a good point. I agree that that response code is a bit confusing. Let me think about what a more appropriate code might be.